In [35]:
import pandas as pd
import numpy as np
from sklearn.linear_model import LogisticRegression
from sklearn.svm import SVC
from sklearn.ensemble import RandomForestClassifier
from sklearn.tree import DecisionTreeClassifier
from sklearn.model_selection import cross_val_score

In [36]:
from sklearn.datasets import load_iris
iris = load_iris()

**logistic regression**

In [37]:
l_scores = cross_val_score(LogisticRegression(max_iter=1000), iris.data, iris.target, cv=10)
l_scores

array([1.        , 0.93333333, 1.        , 1.        , 0.93333333,
       0.93333333, 0.93333333, 1.        , 1.        , 1.        ])

In [38]:
np.average(l_scores)

np.float64(0.9733333333333334)

**Decision Tree**

In [39]:
d_scores = cross_val_score(DecisionTreeClassifier(), iris.data, iris.target, cv=10)
d_scores

array([1.        , 0.93333333, 1.        , 0.93333333, 0.93333333,
       0.86666667, 0.93333333, 1.        , 1.        , 1.        ])

In [40]:
np.average(d_scores)

np.float64(0.96)

**Support Vector Machine (SVM)**

In [41]:
svm_scores = cross_val_score(SVC(), iris.data, iris.target, cv=10)
svm_scores

array([1.        , 0.93333333, 1.        , 1.        , 1.        ,
       0.93333333, 0.93333333, 0.93333333, 1.        , 1.        ])

In [42]:
np.average(svm_scores)

np.float64(0.9733333333333334)

**Random Forest**

In [43]:
rf_scores = cross_val_score(RandomForestClassifier(n_estimators=40), iris.data, iris.target, cv=10)
rf_scores

array([1.        , 0.93333333, 1.        , 0.93333333, 0.93333333,
       0.93333333, 0.93333333, 1.        , 1.        , 1.        ])

In [44]:
np.average(rf_scores)

np.float64(0.9666666666666666)

best score so far from logistic and svm = 0.9733333333333334